sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java

```diff
@@ -86,6 +86,12 @@ static ParentBasedSamplerBuilder parentBasedBuilder(Sampler root) {
    * @throws IllegalArgumentException if {@code ratio} is out of range
    */
   static Sampler traceIdRatioBased(double ratio) {
+    if (ratio == 0.0d) {
+      return alwaysOff();
+    }
+    if (ratio == 1.0d) {
+      return alwaysOn();
```

Comment:
This changes the implementation of `getDescription()` in a way that would be surprising. I.e. you think you're adding a traceIdRatioSampler, but because of this, the `getDescription()` (and `toString()`)  [indicate that you have an alwaysOnSampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/AlwaysOnSampler.java#L31-L39).

Side note: I think its silly samplers have a description at all 🤷. Its a weird [part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#getdescription) which I don't believe has any use.

Reply:
> Its a weird [part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#getdescription)

👍

Reply:
I'm not sure it's all that surprising. If you put in 1.0, I think it's reasonable to get a sampler description that says "always on". I wouldn't block on this.

Reply:
> I think it's reasonable to get a sampler description that says "always on"

I agree it's reasonable, but I think it goes against the spec language 🤷‍♂️

> [The `TraceIdRatioBased` GetDescription MUST return a string of the form `"TraceIdRatioBased{RATIO}"`
with `RATIO` replaced with the Sampler instance's trace sampling ratio
represented as a decimal number.](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#traceidratiobased-sampler-description)